### DISCUSSION TITLE
2024-10-21-r1

### DISCUSSION THREAD

**Discussion by Zoey2936** [ZoeyVid/NPMplus#1176] [Announcements]
# Breaking changes in host names and custom acme servers, fixed upstreams CVEs, please report any issues you find

## What's Changed
- merge upstream (including CVE fixes - shell/command injection while creating certificates - permission to create certs was required to exploit)
  - Note: upstream fixed this by adding regex checks in API endpoint (is included), this may break new hosts/your hosts if you update them, and you are using regex inside your host names
  - Note: I added a second fix by switching from ('child_process').exec to require('child_process').execFile, this should prevent this kind of injection completely, so the regex change from upstream should be safe to revert if needed
  - I also fixed some small upstream bugs which I already noticed while testing, so there are probably more through new regex checks
- dep updates
- generate nginx access lists using node module instead of apache2-utils
- for new instances, fix appsec file upload: increase appsec timeouts, I recommend that you create a backup of your `/opt/npm/etc/crowdsec/crowdsec.conf` config file, delete it, then restart NPMplus (so the file gets recreated) and you then configure it again, so you have the new timeouts
- reload nginx after certbot-ocsp-fetcher.sh finished
- custom certbot.ini support REMOVED, if you want to use a custom acme server please do this using the new envs you can find in the compose.yaml
- fix disabling ipv6
- nginx/custom is now nginx_custom (from sub folder to new folder)
- you no longer need to enter your email while creating certs (please do this in compose.yaml)
- DNS propagation delay support removed
- fixed duplicating '#' in nginx.conf in rootless mode
- allow backend to compress
- improve default ssl_ecdh_curve (enable X25519MLKEM768)
- also disable proxy_request_buffering if you disable proxy buffering using env
- limit mine types to compress (text/images/auido) - if you have ideas if something is missing or should not be compressed please open a discussion (currently: text/html text/css text/javascript text/xml application/atom+xml application/rss+xml text/markdown text/mathml text/plain text/vnd.sun.j2me.app-descriptor text/vnd.wap.wml text/x-component application/json application/xhtml+xml application/xspf+xml font/woff font/woff2 image/avif image/bmp image/png image/svg+xml image/tiff image/vnd.wap.wbmp image/webp image/x-icon image/x-jng audio/midi audio/mpeg audio/ogg audio/x-m4a audio/x-realaudio)

### How to update
- Read the changes above
- Pull the zoeyvid/npmplus:latest image
- apply possible changes that maybe effect you from above to your compose.yaml/NPMplus
- redeploy the compose stack
- report any issues you find

**Full Changelog**: https://github.com/ZoeyVid/NPMplus/compare/2024-10-05-r1...2024-10-21-r1

<hr /><em>This discussion was created from the release <a href='https://github.com/ZoeyVid/NPMplus/releases/tag/2024-10-21-r1'>2024-10-21-r1</a>.</em>

**Comment by tobias86aa** [comment]
Seems to be stuck doing following task: npmplus  | [Global   ] › ⬤  debug     CMD: certbot-ocsp-fetcher.sh -c /data/tls/certbot -o /data/tls/certbot/live --no-reload-webserver --quiet

Nothing seems to be happen and the container report as unhealthy.

**Reply by Zoey2936** [reply]
can you please run:  `certbot-ocsp-fetcher.sh -c /data/tls/certbot -o /data/tls/certbot/live --no-reload-webserver -v` inside the container please and send me the ourput?

**Reply by tobias86aa** [reply]
`certbot-ocsp-fetcher.sh -c /data/tls/certbot -o /data/tls/certbot/live --no-reload-webserver -v
Running in stand-alone mode...

Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
LINEAGE RESULT  REASON
npm-10  failed to update        leaf certificate expired
npm-11  not updated     valid staple file on disk
npm-12  failed to update        leaf certificate expired
npm-13  failed to update        leaf certificate expired
npm-14  failed to update        leaf certificate expired
npm-15  not updated     valid staple file on disk
npm-16  not updated     valid staple file on disk
npm-17  not updated     valid staple file on disk
npm-19  failed to update        leaf certificate expired
npm-26  failed to update        leaf certificate expired
npm-27  failed to update        leaf certificate expired
npm-28  not updated     valid staple file on disk
npm-3   failed to update        leaf certificate expired
npm-35  not updated     valid staple file on disk
npm-36  not updated     valid staple file on disk
npm-38  not updated     valid staple file on disk
npm-4   not updated     valid staple file on disk
npm-6   not updated     valid staple file on disk
npm-7   not updated     valid staple file on disk
npm-9   not updated     valid staple file on disk


Install the BSD utility `column` for properly formatted output.
If the version of `column` supports the `--output-separator` flag,
the output will be formatted as TSV.`

**Reply by Zoey2936** [reply]
it should be fixed in the develop, can you please retry with the :develop tag, please?

**Reply by tobias86aa** [reply]
Same. I try the developer before i created this. And i try it now again it stuck on the same point.

**Reply by tobias86aa** [reply]
`docker1:/app# certbot-ocsp-fetcher.sh -c /data/tls/certbot -o /data/tls/certbot/live --no-reload-webserver -v
Running in stand-alone mode...

Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
Response verify OK
LINEAGE RESULT  REASON
npm-10  failed to update        leaf certificate expired
npm-11  not updated     valid staple file on disk
npm-12  failed to update        leaf certificate expired
npm-13  failed to update        leaf certificate expired
npm-14  failed to update        leaf certificate expired
npm-15  not updated     valid staple file on disk
npm-16  not updated     valid staple file on disk
npm-17  not updated     valid staple file on disk
npm-19  failed to update        leaf certificate expired
npm-26  failed to update        leaf certificate expired
npm-27  failed to update        leaf certificate expired
npm-28  not updated     valid staple file on disk
npm-3   failed to update        leaf certificate expired
npm-35  not updated     valid staple file on disk
npm-36  not updated     valid staple file on disk
npm-38  not updated     valid staple file on disk
npm-4   not updated     valid staple file on disk
npm-6   not updated     valid staple file on disk
npm-7   not updated     valid staple file on disk
npm-9   not updated     valid staple file on disk


Install the BSD utility `column` for properly formatted output.
If the version of `column` supports the `--output-separator` flag,
the output will be formatted as TSV.`

**Reply by Zoey2936** [reply]
sorry, I was to fast the fix is still building: https://github.com/ZoeyVid/NPMplus/actions/runs/11454374356/job/31868421861

**Reply by Zoey2936** [reply]
please try again

**Reply by tobias86aa** [reply]
Thanks now it is working :)

**Reply by Zoey2936** [reply]
I will create a new release

**Reply by gadget78** [reply]
oddly, all the links within /usr/local/nginx/conf/nginx.conf, is trying to use the wrong path, somehow it uses a _ and not a / .. 

`[emerg] open() "/data/nginx_custom/root_top.conf" failed (2: No such file or directory) in /usr/local/nginx/conf/nginx.conf:9`

where `nginx_custom`  should be `nginx/custom` 
i do have `- "IPRT=48"` set in the dockerfile, if that makes a difference ? 

EDIT .. 
just see in the info, you do say "nginx/custom is now nginx_custom (from sub folder to new folder)"
think its causing issues somewhere, as mine doesnt start, and has a error of.... 
`[emerg] open() "/data/nginx_custom/root_top.conf" failed (2: No such file or directory) in /usr/local/nginx/conf/nginx.conf:9`

**Reply by Zoey2936** [reply]
this is migrated:
https://github.com/ZoeyVid/NPMplus/blob/90c3ec73350773427587c9e7af341370b2babbe9/rootfs/usr/local/bin/start.sh#L445
and even if not, it is recreated:
https://github.com/ZoeyVid/NPMplus/blob/90c3ec73350773427587c9e7af341370b2babbe9/rootfs/usr/local/bin/start.sh#L461
https://github.com/ZoeyVid/NPMplus/blob/90c3ec73350773427587c9e7af341370b2babbe9/rootfs/usr/local/bin/start.sh#L575...589

**Reply by gadget78** [reply]
this is using a new install, something must of gone wrong with the install, as the data folder is empty... 

not sure why the drastic move,
from 
/data/nginx/custom/
to
/data/nginx_custom/

i will wait a bit, an re-try

**Reply by Zoey2936** [reply]
to make clearer what file can be edited by you and which are generated by npmplus and should not be changed by you

**Reply by gadget78** [reply]
o i see now, its only the "custom" folder thats been changed .. 

an i found the issue i was having, its because you have updated the compose.yml file 
and by default `      - "ACME_EMAIL=your-email" ` is not set.
so by default it fails because of this. (after i read the docker logs, i realised)

**Reply by Zoey2936** [reply]
ACME_EMAIL has no influence on this

**Reply by gadget78** [reply]
the default compose.yml
https://raw.githubusercontent.com/ZoeyVid/NPMplus/refs/heads/develop/compose.yaml
has on line 13
      - "ACME_EMAIL=your-email" # email address which should be used for acme, ......

so with this in place, it fails with "ACME_EMAIL needs to contains @."

and so fails to load...

EDIT, sorry this doesnt have any connection with the OP now ... thought this was a "any issues" thread ..

**Reply by Zoey2936** [reply]
it only throws this error if you set it to somnething invalid, you can also keep it empty and also it has nothing to do with the nginx_custom folder

**Reply by gadget78** [reply]
indeed, it does only throw this error when its set with something invalid,
which is how you have left the compose.yml by default... 
so "out of the box" it fails. (if you had commented/# this out by default, it will then work "out of the box")

and indeed, this has nothing to do with "nginx_custom" folder, sorry these where two different issues...
i initially didn't realise, and hadn't migrated, and then found the email error too.... 

i have fully replied here to hopefully make things a little clearer, for people reading along, as it may help someone.
(i have found another issue on newer a build, with connection to this email issues, i will post on the newer build thread after this)

**Reply by Zoey2936** [reply]
yes the fail is wanted since it should be changed, like TZ
